src/test/java/de/tum/in/www1/artemis/exercise/ExerciseIntegrationTest.java

```diff
@@ -206,57 +231,140 @@ void testGetExercise() throws Exception {
+        }
+        else {
+            assertThat(exercise.getExampleSolution()).as("Sample solution not present").isNull();
+        }
+    }
+
+    private void assertModelingExercise(ModelingExercise exercise, DiagramType diagramType, String exampleSolutionModel, String exampleSolutionExplanation) {
+        assertThat(exercise.getDiagramType()).as("Diagram type was set correctly").isEqualTo(diagramType);
+        if (exampleSolutionModel != null) {
+            assertThat(exercise.getExampleSolutionModel()).as("Sample solution model was set correctly").isEqualTo(exampleSolutionModel);
+        }
+        else {
+            assertThat(exercise.getExampleSolutionModel()).as("Sample solution model not present").isNull();
+        }
+        if (exampleSolutionExplanation != null) {
+            assertThat(exercise.getExampleSolutionExplanation()).as("Sample solution explanation was set correctly").isEqualTo(exampleSolutionExplanation);
+        }
+        else {
+            assertThat(exercise.getExampleSolutionExplanation()).as("Sample solution explanation not present").isNull();
+        }
+    }
+
+    private void assertProgrammingExercise(ProgrammingExercise exercise, boolean projectKey, String templateRepositoryUri, String solutionRepositoryUri, String testRepositoryUri,
+            String templateBuildPlanId, String solutionBuildPlanId) {
+        if (projectKey) {
+            assertThat(exercise.getProjectKey()).as("Project key was set").isNotNull();
+        }
+        else {
+            assertThat(exercise.getProjectKey()).as("Project key not present").isNull();
+        }
+        if (templateRepositoryUri != null) {
+            assertThat(exercise.getTemplateRepositoryUri()).as("Template repository uri was set correctly").isEqualTo(templateRepositoryUri);
+        }
+        else {
+            assertThat(exercise.getTemplateRepositoryUri()).as("Template repository uri not present").isNull();
+        }
+        if (solutionRepositoryUri != null) {
+            assertThat(exercise.getSolutionRepositoryUri()).as("Solution repository uri was set correctly").isEqualTo(solutionRepositoryUri);
+        }
+        else {
+            assertThat(exercise.getSolutionRepositoryUri()).as("Solution repository uri not present").isNull();
+        }
+        if (testRepositoryUri != null) {
+            assertThat(exercise.getTestRepositoryUri()).as("Test repository uri was set correctly").isEqualTo(testRepositoryUri);
+        }
+        else {
+            assertThat(exercise.getTestRepositoryUri()).as("Test repository uri not present").isNull();
+        }
+        if (templateBuildPlanId != null) {
+            assertThat(exercise.getTemplateBuildPlanId()).as("Template build plan was set correctly").isEqualTo(templateBuildPlanId);
+        }
+        else {
+            assertThat(exercise.getTemplateBuildPlanId()).as("Template build plan not present").isNull();
+        }
```

Comment:
We have a lot of very similar _repetitive_ structures everywhere around here. Can we create a method for that, too?

For example, a method that would allow us to replace each
```java
if (solutionRepositoryUri != null) {
    assertThat(exercise.getSolutionRepositoryUri()).as("Solution repository uri was set correctly").isEqualTo(solutionRepositoryUri);
} else {
    assertThat(exercise.getSolutionRepositoryUri()).as("Solution repository uri not present").isNull();
}
```

with this:
        
```java
 assertEqualOrNull(solutionRepositoryUri, exercise.getSolutionRepositoryUri(), "Solution repository uri");
```

The new method would then be something along these lines:
        
```java
private void assertEqualOrNull(Object expected, Object actual, String entityName) {
    if (expected == null) {
        assertThat(actual).as(entityName + " not present").isNull();
    } else {
        assertThat(actual).as(entityName + " was set correctly").isEqualTo(expected);
    }
}
```